src/LinAlg/hiopLinSolverSymSparseMA57.hpp

```diff
@@ -156,14 +207,115 @@ class hiopLinSolverSymSparseMA57: public hiopLinSolverSymSparse
 
 public:
 
-  /** called the very first time a matrix is factored. Allocates space
-   * for the factorization and performs ordering */
+  /** 
+   * Called the very first time a matrix is factorized, this method allocates space for the 
+   * factorization and performs ordering. 
+   */
   virtual void firstCall();
-//  virtual void diagonalChanged( int idiag, int extent );
 
   // increase pivot tolarence
   virtual bool increase_pivot_tol();
 
+  };
+
+/** 
+ * MA57 solver class that takes CSR sparse input and offers the boilerplate to copy this into the internal 
+ * triplet matrix used with MA57 API. 
+ * 
+ * The CSR matrix is understood to be symmetric. The underlying CSR storage can contain all the nonzero entries 
+ * or only the lower triangular part. In both cases, this class will copy ONLY the lower triangular entries to 
+ * the underlying triplet storage.
+ */  
```

Comment:
Do we have a way/function to check if the input CSR matrix is `upper` triangular, which will fail all the methods below?